controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -339,12 +339,28 @@ func (r *BareMetalHostReconciler) credentialsErrorResult(err error, request ctrl
 }
 
 // hasRebootAnnotation checks for existence of reboot annotations and returns true if at least one exist
-func hasRebootAnnotation(host *metal3v1alpha1.BareMetalHost) bool {
-	for annotation := range host.Annotations {
+func hasRebootAnnotation(host *metal3v1alpha1.BareMetalHost) (hasReboot, hardMode bool) {
+	for annotation, value := range host.GetAnnotations() {
 		if isRebootAnnotation(annotation) {
-			return true
+			hasReboot = true
+			if isHardReboot(value) {
+				hardMode = true
+			}
```

Comment:
A break here would make sense.

Reply:
@dhellmann you mean break once it's found the first reboot annotation? If so, I considered that, but then I remembered that there may be multiple clients here that could set a reboot annotation. First one may be soft, the second one hard, in which we'd want the hard one to be honoured.

Reply:
Oh, good point, yes. Maybe a comment explaining why we don't want to break, to help *cough* "future reviewers" then. :-)

Reply:
Sounds good :-)